src/sentry/grouping/strategies/base.py

```diff
@@ -468,7 +455,7 @@ def call_with_variants(
             component = rv_variants[variant.lstrip("!")]
 
         if component is None:
-            continue
+            continue  # type:ignore[unreachable]
```

Comment:
this should not be ignored -- some type is wrong here then -- mypy is saying that component cannot be none so `rv_variants` must have the wrong type

Reply:
Yes exactly, the type does not include `None`. I tried to add it but it was immediately complaining someplace else, and I wasn’t prepared to fix all the fallout, so this is just a temporary fix at best.

Reply:
I would prefer we fix it properly rather than ignoring the diagnostic since we control the code here

`type: ignore` generally should only be used for bugs in the type checker or terrible hacks -- this doesn't really fall into either of those categories (the checker is correct to tell us this is a mistake here)

Reply:
I believe the issue here is simply that `rv_variants` has type `dict[str, GroupingComponent]` and so `component` is either a `GroupingComponent` or there's a key error. It can't be `None` because the dict isn't accessed with `get`.